x/staking/keeper/grpc_query_test.go

```diff
@@ -288,36 +288,45 @@ func (suite *KeeperTestSuite) TestGRPCQueryDelegatorDelegations() {
 		msg      string
 		malleate func()
 		expPass  bool
+		expErr   bool
 	}{
 		{"empty request",
 			func() {
 				req = &types.QueryDelegatorDelegationsRequest{}
 			},
 			false,
+			true,
 		}, {"invalid request",
 			func() {
 				req = &types.QueryDelegatorDelegationsRequest{DelegatorAddr: addrs[4].String()}
 			},
 			false,
+			false,
 		},
 		{"valid request",
 			func() {
 				req = &types.QueryDelegatorDelegationsRequest{DelegatorAddr: addrAcc.String(),
 					Pagination: &query.PageRequest{Limit: 1, CountTotal: true}}
 			},
 			true,
+			false,
 		},
 	}
 
 	for _, tc := range testCases {
 		suite.Run(fmt.Sprintf("Case %s", tc.msg), func() {
 			tc.malleate()
 			res, err := queryClient.DelegatorDelegations(gocontext.Background(), req)
-			if tc.expPass {
+			if tc.expPass && !tc.expErr {
+				suite.NoError(err)
+				suite.NotNil(res.Pagination.NextKey)
 				suite.Equal(uint64(2), res.Pagination.Total)
 				suite.Len(res.DelegationResponses, 1)
 				suite.Equal(1, len(res.DelegationResponses))
 				suite.Equal(sdk.NewCoin(sdk.DefaultBondDenom, delegation.Shares.TruncateInt()), res.DelegationResponses[0].Balance)
```

Comment:
> I think this condition is confusing a bit. Instead of this extra condition, can you add the `expectResult` in testCases struct and pass the expected result there, hence there will be only two conditions (either pass or fail).
Here, we are already checking `DelegationResponse`, so would we need `expectResult` in testCases struct ?